spec/index.bs

```diff
@@ -2510,146 +2504,227 @@ Issue(https://github.com/gpuweb/gpuweb/issues/851): consider making `textureComp
+        <td rowspan=3>{{GPUBindGroupLayoutEntry/texture}}
+        <td rowspan=3>{{GPUTextureView}}
+        <td>{{GPUTextureType/"filterable"}}
         <td>[=internal usage/constant=]
     <tr>
-        <td>{{GPUBindingType/"readonly-storage-texture"}}
-        <td>{{GPUTextureView}}
+        <td>{{GPUTextureType/"unfilterable"}}
+        <td>[=internal usage/constant=]
+    <tr>
+        <td>{{GPUTextureType/"multisample"}}
+        <td>[=internal usage/constant=]
+
+    <tr>
+        <td rowspan=2>{{GPUBindGroupLayoutEntry/storageTexture}}
+        <td rowspan=2>{{GPUTextureView}}
+        <td>{{GPUStorageTextureType/"readonly"}}
         <td>[=internal usage/storage-read=]
     <tr>
-        <td>{{GPUBindingType/"writeonly-storage-texture"}}
-        <td>{{GPUTextureView}}
+        <td>{{GPUStorageTextureType/"writeonly"}}
         <td>[=internal usage/storage-write=]
 </table>
 
+<div algorithm>
+    To get the <dfn abstract-op>layout entry binding type</dfn> in a given {{GPUBindGroupLayoutEntry}} |entry|:
+
+    1. If |entry|.{{GPUBindGroupLayoutEntry/buffer}} is not `undefined`:
+        1. Return |entry|.{{GPUBindGroupLayoutEntry/buffer}}.{{GPUBufferLayoutEntry/type}}.
+    1. If |entry|.{{GPUBindGroupLayoutEntry/sampler}} is not `undefined`:
+        1. Return |entry|.{{GPUBindGroupLayoutEntry/sampler}}.{{GPUSamplerLayoutEntry/type}}.
+    1. If |entry|.{{GPUBindGroupLayoutEntry/texture}} is not `undefined`:
+        1. Return |entry|.{{GPUBindGroupLayoutEntry/texture}}.{{GPUTextureLayoutEntry/type}}.
+    1. If |entry|.{{GPUBindGroupLayoutEntry/storageTexture}} is not `undefined`:
+        1. Return |entry|.{{GPUBindGroupLayoutEntry/storageTexture}}.{{GPUStorageTextureLayoutEntry/type}}.
+</div>
+
+<script type=idl>
+enum GPUBufferType {
+    "uniform",
+    "storage",
+    "readonly-storage",
+};
+
+dictionary GPUBufferLayoutEntry {
+    GPUBufferType type = "uniform";
+    boolean hasDynamicOffset = false;
+    GPUSize64 minBufferBindingSize = 0;
+};
+</script>
+
+{{GPUBufferLayoutEntry}} dictionaries have the following members:
+
+<dl dfn-type=dict-member dfn-for=GPUBufferLayoutEntry>
+    : <dfn>type</dfn>
+    ::
+        Indicates the type required for buffers bound to this bindings.
+
+    : <dfn>hasDynamicOffset</dfn>
+    ::
+        Indicates whether this binding requires a dynamic offset.
+
+    : <dfn>minBufferBindingSize</dfn>
+    ::
+        May be used to indicate the minimum buffer binding size.
+</dl>
+
+<script type=idl>
+enum GPUSamplerType {
+    "filtering",
+    "non-filtering",
+    "comparison",
+};
+
+dictionary GPUSamplerLayoutEntry {
+    GPUSamplerType type = "filtering";
+};
+</script>
+
+{{GPUSamplerLayoutEntry}} dictionaries have the following members:
+
+<dl dfn-type=dict-member dfn-for=GPUSamplerLayoutEntry>
+    : <dfn>type</dfn>
+    ::
+        Indicates the required type of a sampler bound to this bindings.
+</dl>
+
+<script type=idl>
+enum GPUTextureType {
+    "filterable",
+    "unfilterable",
```

Comment:
I think there is a problem with this approach. When using non-float textures, it would be nice if the users could rely on the default `GPUTextureType` and not specify it, because clearly it's redundant. Non-float textures can't be sampled in WGSL, so they naturally aren't filterable (and this term shouldn't even apply...). So we can't default this to `filterable`.

How about this strawman:
```webidl
enum GPUTextureType {
  "float",
  "unfilterable-float",
  "depth-float",
  "sint",
  "uint",
};
dictionary GPUTextureLayoutEntry {
    GPUTextureType type = "float";
    GPUTextureViewDimension viewDimension = "2d";
    bool multisampled = false;
};
```

Names are subject to change, of course :) The benefit of this is no need to specify redundant info for integer textures.

Reply:
Shouldn't be a problem, but I'll leave it to others a bit better versed in the underlying APIs to greenlight this before I make the change.

Reply:
+1 to @kvark's suggestion. I think we recently (re-)realized that D3D12 prevented any form of sampling of integer textures so the filterable / unfilterable distinction is only for float textures.

The names `depth-float` and `GPUTextureType` seem a bit generic though, so I'd be open to bikeshed them. (though that could be in a follow-up if this is forcing too many iterations)